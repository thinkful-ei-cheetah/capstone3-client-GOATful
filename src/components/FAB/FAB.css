button.fab {
  background: #e53030;
  border: 3px solid #fff;
  border-radius: 100%;
  box-shadow: 0 -2px 0 3px #c91919 inset, 0 5px 5px rgba(65, 8, 8, 0.17), 0 15px rgba(255, 255, 255, 0.25) inset;
  cursor: pointer;
  display: inline-block;
  position:fixed;
	width:70px;
	height:70px;
	bottom:40px;
	right:40px;
	/* background-color:#c4302b; */
	color:#FFF;
	border-radius:50px;
	text-align:center;
  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
  overflow: hidden;
  z-index: 1;
  outline: none;
}

button.fab:hover {
  background: #333;
  box-shadow: 0 -2px 0 3px #1a1a1a inset, 0 5px 5px rgba(0, 0, 0, 0.17), 0 15px rgba(255, 255, 255, 0.32) inset;
}

button.fab:after {
  content: "";
  background: rgb(208, 81, 81);
  display: block;
  position: absolute;
  padding-top: 300%;
  padding-left: 350%;
  margin-left: -20px!important;
  margin-top: -120%;
  opacity: 0;
  transition: all 0.8s
}

button.fab:active:after {
  padding: 0;
  margin: 0;
  opacity: 1;
  transition: 0s
}

.videos-page > .fab {
  bottom: 80px;
}