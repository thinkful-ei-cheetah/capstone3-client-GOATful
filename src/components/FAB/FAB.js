import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'
import './FAB.css'

export default function FAB(props) {
  return (
    <button className='fab' onClick={props.onClick}>
      <FontAwesomeIcon 
        icon={faPlus} 
        className='fab-icon'
        onClick={props.onClick}
      />
    </button>
  )
}
