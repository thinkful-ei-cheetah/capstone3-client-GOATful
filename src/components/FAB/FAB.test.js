import React from 'react';
import ReactDOM from 'react-dom';
import FAB from './FAB';
import renderer from 'react-test-renderer';

describe('FAB Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< FAB />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<FAB/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
