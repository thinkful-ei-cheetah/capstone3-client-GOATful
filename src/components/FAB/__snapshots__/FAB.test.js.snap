// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB Component renders the UI as expected 1`] = `
<button
  className="fab"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-plus fa-w-14 fab-icon"
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
